### PR TITLE
BMC Dump fixes and enhancements

### PR DESCRIPTION
**Pull request by geissonator**

### CODE REVIEW

File: geissonator/openbmc-events/openbmc-events [hunk 1]
```diff
@@ -78,7 +78,10 @@ class BMC:
         return dumps
     
     def create_dump(self):
-        r = self.session.post(self.url + 'xyz/openbmc_project/dump/action/CreateDump')
+        r = self.session.post(self.url + 'xyz/openbmc_project/dump/action/CreateDump',
+                              headers={'Content-Type': 'application/json'},
+                              data='{"data":[]}',
+                              verify=False)
```
Comment (amboar) [inline geissonator/openbmc-events/openbmc-events]: I don't at all like the `verify=False` part. We should let the user choose.

File: geissonator/openbmc-events/openbmc-events [hunk 2]
```diff
@@ -117,8 +120,12 @@ def do_clear_all_events(args):
 
 def do_list_dumps(args):
     s = BMC(server=args.server)
-    for e in s.list_dumps():
-        print(e)
+    try:
+        for e in s.list_dumps():
+            print(e)
+    except:
+        # ignore if we get nothing back
```
Comment (amboar) [inline geissonator/openbmc-events/openbmc-events]: This is a bit wild. If there are no dumps, we should just return an empty list, not throw an exception. Exceptions should be used for exceptional circumstances, and not having any dumps is not one.

File: geissonator/openbmc-events/openbmc-sfw [hunk 3]
```diff
@@ -6,6 +6,9 @@ import tarfile
 import time
 import json
 
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```
Comment (amboar) [inline geissonator/openbmc-events/openbmc-sfw]: We should let the user choose. See what I did for `openbmc-sfw` (which I pushed without your review accidentally).

File: geissonator/openbmc-events/openbmc-events (hunk 1 above)
Comment (geissonator) [reply geissonator/openbmc-events/openbmc-events]: Yeah, that's not even needed, just a copy/paste I guess from some other one's.

File: geissonator/openbmc-events/openbmc-events (hunk 2 above)
Comment (geissonator) [reply geissonator/openbmc-events/openbmc-events]: Makes sense, done

File: geissonator/openbmc-events/openbmc-sfw (hunk 3 above)
Comment (geissonator) [reply geissonator/openbmc-events/openbmc-sfw]: Done

Comment (geissonator) [reply geissonator/openbmc-events/openbmc-sfw]: Well, I did this in openbmc-events at least.  Looks like you already did openbmc-sfw. Guess I need to get this commit stream rebased.

**Pull request merged**